TWLight/users/oauth.py

```diff
@@ -324,10 +327,31 @@ def get(self, request, *args, **kwargs):
                 )
                 raise PermissionDenied
 
-            local_redirect = _localize_oauth_redirect(redirect)
+            # Create a QueryDict from the 'get' session dict.
+            query_dict = QueryDict(urlencode(request.session["get"]), mutable=True)
+            # Pop the 'next' parameter out of the QueryDict.
+            next = query_dict.pop("next")
+            # Set the return url to the value of 'next'. Basic.
+            return_url = next[0]
+
+            if "oclc" in return_url or "ezproxy" in return_url:
```

Comment:
Quick question on this - is there a particular reason to limit this behaviour to proxy links? I'm also thinking about other login-required pages, for example if someone links directly to /my_library, where this would be helpful.

Reply:
Solved!